API/Data/Repositories/VolumeRepository.cs

```diff
@@ -218,7 +219,11 @@ private async Task AddVolumeModifiers(int userId, IReadOnlyCollection<VolumeDto>
         {
             foreach (var c in v.Chapters)
             {
-                c.PagesRead = userProgress.Where(p => p.ChapterId == c.Id).Sum(p => p.PagesRead);
+                var progress = userProgress.SingleOrDefault(p => p.ChapterId == c.Id);
+                if (progress == null) continue;
+
+                c.PagesRead = progress.PagesRead;
+                c.ProgressLastModifiedUtc = progress.LastModifiedUtc;
```

Comment:
You can skip the null check and use null coalescing instead.

c.ProgressLastModifiedutc = progress?.LastModifiedUtc;

Add the non Utc field to be consistent.

Reply:
ok